arch/arc/Kconfig

```diff
@@ -263,7 +263,7 @@ config CODE_DENSITY
 config ARC_HAS_ACCL_REGS
 	bool "Reg Pair ACCL:ACCH (FPU and/or MPY > 6)"
 	default y if CPU_HS3X
-	default y if FPU
+	default y if CPU_HAS_FPU
```

Comment:
BTW, the default FPU usage is without FPU_SHARING - we assume that we use FPU only in one thread, so we don't need ARC_HAS_ACCL_REGS if we have FPU by default (at least on ARC HS).

I guess we are enabling ARC_HAS_ACCL_REGS because we assuming that ACCL_REGS would be used by MPY instructions. Probably we need to add comment here that if we don't need to enable it if our system doesn't support MPY instructions / we are 100% sure that we don't use them.

Reply:
Triggered by this, I checked and actually believe FPU doesn't always mean the accl registers are there. Instead, they are only available if fused multiply & add/sub is configured in the hardware.
I think this default line can simply be removed and left to SoC configuration.